### ISSUE TITLE
Suggest `Did you mean 'typeof Foo'` when using value Foo as a type

### ISSUE THREAD

**Issue by mohsen1** [microsoft/TypeScript#28975]
```ts
type Foo<T> = {}

const Something = []

type FooSomething = Foo<Something>
//                      ^
//                      Can not find name "Something"               
```

Instead, suggest that `Something` is in the value space and user possibly meant `typeof Something`.
## Search Terms

Error messages, type space, generics 

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by DanielRosenwasser** [comment]
I think this is more general than a type argument; probably as a type in general.

**Comment by DanielRosenwasser** [comment]
@weswigham has some concerns regarding things like the following

```ts
type FooInstance = { }

function Foo(): FooInstance {

}

// you probably don't want this to be corrected to 'typeof Foo'
type FooAlias = Foo;
```

**Comment by s-lambert** [comment]
But [that code doesn't seem to work](https://www.typescriptlang.org/play/index.html#src=type%20FooInstance%20%3D%20%7B%7D%3B%0D%0A%0D%0Afunction%20Foo()%3A%20FooInstance%20%7B%0D%0A%20%20%20%20return%20%7B%7D%3B%0D%0A%7D%0D%0A%0D%0A%2F%2F%20you%20probably%20don't%20want%20this%20to%20be%20corrected%20to%20'typeof%20Foo'%0D%0Atype%20FooAlias%20%3D%20Foo%3B%0D%0A), it won't compile because you do need `typeof Foo` to assign it to a type.

**Comment by cpplearner** [comment]
Perhaps it's more likely that `InstanceType<typeof Foo>` is intended?

**Comment by suhailsinghbains** [comment]
Hi, Can I please work on this issue, I am new here

**Comment by jtbandes** [comment]
@amcasey Should this issue be closed now as #37359 is merged? Looks like #32543 would have auto-closed this ticket but it was abandoned in favor of #37359.

**Comment by weswigham** [comment]
Ah, yep.
